test/unit/runner.spec.js

```diff
@@ -487,6 +487,32 @@ describe('Runner', function() {
         });
         runner.failHook(hook, err);
       });
+
+      it('should not hang if the error message is ridiculously long', function(done) {
+        var hook = new Hook();
+        var message = [];
+        // mock a long message
+        for (var i = 0; i < 20000; i++) {
+          var line = stack[i % stack.length];
+          message[i] = line[Math.floor(Math.random() * line.length)];
+        }
```

Comment:
OK. Think I'm good with the regex change to code in "lib/utils.js". But uncomfortable with your test here, which seems at odds with your issue. Why doesn't it just create a multi-line `AssertionError.message` and replace `stack[0]`? The `for` loop code seems... excessive-? to that task, and its contents are non-reproducible.

Reply:
@plroebuck ah, good catch. There actually are two factors that might slow down the regex test, a long single line (>= 40k takes ~1600ms), or a large message with multiple lines. The test case I constructed was trying to mimic the first case but it was done incorrectly. I've pushed the fix.

The `for` loop is the simplest way I can come up with to create a long and non-repeating text. Using `fs.readFileSync(path.join(__dirname, '../mocha.js'), 'utf8')` suffices the task but I'm not sure this unit test is node.js only or not.

Reply:
ha, turns out I wasn't very wrong. 

```bash
> a = [1,2]
[ 1, 2 ]
> [a]
[ [ 1, 2 ] ]
> [a].concat(3).join('\n')
'1,2\n3'
```

`[message].concat(stack).join(\n')` just yields a large `err.message` with lots of `,`. Anyway, I've updated the test code.

Reply:
Could you add a test case for a large multi-line message to cover _both_ triggers?

Reply:
ah, guess I missed this one. I'm on it.